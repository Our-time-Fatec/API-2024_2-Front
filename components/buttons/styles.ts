import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({

botaoazul:{
    flex: 1,
    color: '#FFFFFF',
    backgroundColor: "#407CE2",
    borderRadius: 40,
    alignItems: 'center',
    padding: 20,
},
textobotaoazul:{
    fontFamily: "Poppins-SemiBold",
    color: "#FFFFFF",
    fontSize: 20
},
botaobranco:{
    width: 300,
    backgroundColor: "#FFFFFF",
    borderRadius: 40,
    alignItems: 'center',
    padding: 20,
    borderColor: '#407CE2',
    borderWidth: 2,
},
textobotaobranco:{
    fontFamily: "Poppins-SemiBold",
    fontSize: 20,
    color: "#407CE2"
}
});

export default styles;