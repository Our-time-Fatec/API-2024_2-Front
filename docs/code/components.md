---
title: components
description: 'Pasta que contém todos os componentes reutilizáveis da aplicação.'
---

# components

A pasta `components` é responsável por armazenar todos os componentes reutilizáveis da aplicação. Esses componentes são fundamentais para a construção da interface do usuário, permitindo a modularização e a reutilização de código.

## Estrutura da Pasta

Abaixo está a estrutura da pasta `components`:

- **colors**: Contém definições de cores utilizadas na aplicação.
- **fonts**: Inclui componentes relacionados a fontes.
- **dieta**: Componentes específicos para a funcionalidade de dietas.
- **alimento**: Componentes relacionados à gestão de alimentos.
- **buttons**: Componentes de botões personalizados.
- **menus**: Componentes para a criação de menus na aplicação.

Cada subpasta pode conter arquivos de implementação e estilos específicos para os componentes, facilitando a organização e manutenção do código.