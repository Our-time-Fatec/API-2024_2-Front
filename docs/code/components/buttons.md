
---
title: buttons
description: 'Contém componentes de botões reutilizáveis para a aplicação.'
---

# buttons

A pasta `buttons` contém componentes de botões reutilizáveis que são utilizados em diferentes partes da aplicação. Esses componentes são projetados para garantir consistência visual e funcionalidade em toda a interface do usuário.

## Estrutura

A pasta `buttons` inclui os seguintes arquivos:

- `index.tsx`: O arquivo principal que exporta os componentes de botão.
- `styles.ts`: Define os estilos dos componentes de botão, permitindo personalização e manutenção centralizada.

## Uso

Os componentes de botão podem ser importados e utilizados em outras partes da aplicação, garantindo uma experiência de usuário coesa. A estrutura modular permite fácil modificação e extensão dos estilos e comportamentos dos botões.

## Contribuição

Para adicionar novos estilos ou tipos de botões, crie novos arquivos dentro da pasta `buttons` e atualize o `index.tsx` para exportar os novos componentes.
