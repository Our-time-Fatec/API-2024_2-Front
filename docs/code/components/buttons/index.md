---
title: index
description: 'Componente de Botões para React Native com estilos personalizados.'
---

# Componente de Botões

Este arquivo contém a implementação de dois componentes de botão para uso em aplicações React Native: `BotaoAzul` e `BotaoBranco`. Ambos os componentes são projetados para serem reutilizáveis e personalizáveis.

## BotaoAzul

O componente `BotaoAzul` é um botão estilizado que utiliza a fonte Poppins. Ele é ideal para ações que requerem destaque.

### Props

- **texto**: `string` - O texto a ser exibido no botão.
- **onPress**: `() => void` - Função a ser chamada quando o botão é pressionado.
- **style**: `StyleProp<ViewStyle>` (opcional) - Estilo adicional para o contêiner do botão.
- **textStyle**: `StyleProp<TextStyle>` (opcional) - Estilo adicional para o texto do botão.

### Exemplo de Uso

```jsx
<BotaoAzul 
  texto="Clique Aqui" 
  onPress={() => console.log('Botão Azul Pressionado')} 
  style={{ margin: 10 }} 
  textStyle={{ color: 'white' }} 
/>
```

## BotaoBranco

O componente `BotaoBranco` é um botão simples, ideal para ações secundárias ou menos destacadas.

### Props

- **texto**: `string` - O texto a ser exibido no botão.
- **onPress**: `() => void` - Função a ser chamada quando o botão é pressionado.
- **style**: `StyleProp<ViewStyle>` (opcional) - Estilo adicional para o contêiner do botão.
- **textStyle**: `StyleProp<TextStyle>` (opcional) - Estilo adicional para o texto do botão.

### Exemplo de Uso

```jsx
<BotaoBranco 
  texto="Cancelar" 
  onPress={() => console.log('Botão Branco Pressionado')} 
  style={{ margin: 10 }} 
  textStyle={{ color: 'black' }} 
/>
```

## Considerações

- O componente `BotaoAzul` carrega fontes utilizando o hook `useFonts` do Expo. Enquanto as fontes não estiverem carregadas, o componente não será exibido.
- O `SplashScreen` é utilizado para evitar que a tela inicial do aplicativo apareça antes que as fontes estejam carregadas.

## Exportação

Os componentes `BotaoAzul` e `BotaoBranco` são exportados para serem utilizados em outras partes da aplicação:

```javascript
export { BotaoAzul, BotaoBranco };
```