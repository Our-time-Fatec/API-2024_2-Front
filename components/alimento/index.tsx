import React, { useState } from "react";
import {
  View,
  Text,
  Button,
  Modal,
  StyleSheet,
  Image,
  TouchableOpacity,
  TextInput,
} from "react-native";
import { IAlimento } from "../../interfaces/IAlimento";
import { requestWithRefresh } from "../../services/api";
import { styles } from "./styles";
import { Picker } from "@react-native-picker/picker";
import { GruposEnum } from "../../screens/cadastrarDieta";

interface AlimentoProps {
  alimento: IAlimento;
  isUserAlimento: boolean;
  isAlimentoConsumido?: boolean;
  onEdit: (id: string) => void;
  onDelete: (id: string) => void;
}

const AlimentoItem: React.FC<AlimentoProps> = ({
  alimento,
  isUserAlimento,
  isAlimentoConsumido,
  onEdit,
  onDelete,
}) => {
  const [modalVisible, setModalVisible] = useState(false);
  const [modalVisibleConsumo, setModalVisibleConsumo] = useState(false);
  const [porcaoInput, setPorcaoInput] = useState("");
  const [quantidadeInput, setQuantidadeInput] = useState("");
  const [nomeGrupo, setNomeGrupo] = useState<string>("");

  const formatDate = (date: Date) => {
    return new Date(date).toLocaleDateString("pt-BR");
  };

  const toggleModal = () => {
    setModalVisible(!modalVisible);
  };

  const toggleModalConsumo = () => {
    setModalVisibleConsumo(!modalVisibleConsumo);
  };

  const handleConsumir = async () => {
    try {
      await requestWithRefresh({
        method: "POST",
        url: `/alimentoConsumido/`,
        data: {
          _id: alimento._id || "",
          porcao: parseFloat(porcaoInput),
          quantidade: parseInt(quantidadeInput),
          nomeGrupo,
        },
      });
      alert("Alimento adicionado como consumido com sucesso!");
      toggleModalConsumo();
    } catch (error) {
      console.log(error);
      alert("Erro ao adicionar alimento consumido.");
    }
  };

  const handleConfirmarConsumo = () => {
    handleConsumir();
  };

  return (
    <View style={styles.card}>
      <TouchableOpacity onPress={toggleModal} style={styles.imageContainer} testID="imageTouchable">
        {alimento.categoriaUrl ? (
          <Image
            source={{ uri: alimento.categoriaUrl }}
            style={styles.image}
            resizeMode="cover"
            testID="alimentoImage"
          />
        ) : (
          <View style={styles.imagePlaceholder} testID="imagePlaceholder">
            <Text style={styles.imagePlaceholderText}>
              Imagem não disponível
            </Text>
          </View>
        )}
      </TouchableOpacity>
      <View style={styles.contentWrapper}>
        <View style={styles.detailsContainer}>
          <Text style={styles.nome} testID="alimentoNome">{alimento.nome}</Text>
          <Text style={styles.info} testID="alimentoPorcao">{`${alimento.porcao}g`}</Text>
          {isAlimentoConsumido && (
            <Text style={styles.info} testID="alimentoGrupo">Refeição: {alimento.nomeGrupo}</Text>
          )}
          <Text style={styles.info} testID="alimentoPreparo">Preparo: {alimento.preparo}</Text>
          <Text style={styles.info} testID="alimentoCategoria">Categoria: {alimento.categoriaNome}</Text>
          {isAlimentoConsumido && (
            <>
              <Text style={styles.info} testID="alimentoQuantidade">Quantidade: {alimento.quantidade}</Text>
              <Text style={styles.info} testID="alimentoDataConsumo">
                Data de consumo:{" "}
                {formatDate(alimento.criadoEm ? alimento.criadoEm : new Date())}
              </Text>
            </>
          )}
        </View>
        <View>
          {isUserAlimento && (
            <View style={styles.buttonContainer}>
              <TouchableOpacity
                style={styles.editButton}
                onPress={() => onEdit(alimento._id ? alimento._id : "")}
                testID="editButton"
              >
                <Text style={styles.closeButtonText}>Editar</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.deleteButton}
                onPress={() => onDelete(alimento._id ? alimento._id : "")}
                testID="deleteButton"
              >
                <Text style={styles.closeButtonText}>Remover</Text>
              </TouchableOpacity>
            </View>
          )}
          {isAlimentoConsumido && (
            <View style={styles.buttonContainer}>
              <TouchableOpacity
                style={styles.deleteButton}
                onPress={() => onDelete(alimento._id ? alimento._id : "")}
                testID="deleteButton"
              >
                <Text style={styles.closeButtonText}>Remover</Text>
              </TouchableOpacity>
            </View>
          )}
          {!isAlimentoConsumido && (
            <View style={styles.buttonContainer}>
              <TouchableOpacity
                style={styles.editButton}
                onPress={toggleModalConsumo}
                testID="consumirButton"
                disabled={isAlimentoConsumido}
              >
                <Text style={styles.closeButtonText}>Consumir</Text>
              </TouchableOpacity>
            </View>
          )}
        </View>
      </View>
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={toggleModal}
        testID="modal"
      >
        <View style={styles.modalContainer}>
          <View style={styles.modalContent}>
            {alimento.categoriaUrl ? (
              <Image
                source={{ uri: alimento.categoriaUrl }}
                style={styles.modalImage}
                resizeMode="cover"
                testID="modalAlimentoImage"
              />
            ) : (
              <View style={styles.imagePlaceholder} testID="modalImagePlaceholder">
                <Text style={styles.imagePlaceholderText}>
                  Imagem não disponível
                </Text>
              </View>
            )}
            <Text style={styles.modalNome} testID="modalAlimentoNome">{alimento.nome}</Text>
            <Text style={styles.modalInfo} testID="modalAlimentoCategoria">
              Categoria: {alimento.categoriaNome}
            </Text>
            <Text style={styles.modalInfo} testID="modalAlimentoPreparo">Preparo: {alimento.preparo}</Text>
            <Text style={styles.modalInfo} testID="modalAlimentoPorcao">Porção: {alimento.porcao}g</Text>
            {isAlimentoConsumido && (
              <>
                <Text style={styles.modalInfo} testID="modalAlimentoQuantidade">
                  Quantidade: {alimento.quantidade}
                </Text>
                <Text style={styles.modalInfo} testID="modalAlimentoDataConsumo">
                  Data de consumo:{" "}
                  {formatDate(alimento.criadoEm ? alimento.criadoEm : new Date())}
                </Text>
              </>
            )}
            <Text style={styles.modalInfo} testID="modalAlimentoValorEnergetico">
              Valor Energético: {alimento.detalhes.valorEnergetico.toFixed(2)} kcal
            </Text>
            <Text style={styles.modalInfo} testID="modalAlimentoProteinas">
              Proteínas: {alimento.detalhes.proteinas.toFixed(2)}g
            </Text>
            <Text style={styles.modalInfo} testID="modalAlimentoCarboidratos">
              Carboidratos: {alimento.detalhes.carboidratos.toFixed(2)}g
            </Text>
            <Text style={styles.modalInfo} testID="modalAlimentoFibras">
              Fibras: {alimento.detalhes.fibras.toFixed(2)}g
            </Text>
            <Text style={styles.modalInfo} testID="modalAlimentoLipidios">
              Lipídios: {alimento.detalhes.lipidios.toFixed(2)}g
            </Text>
            <TouchableOpacity style={styles.closeButton} onPress={toggleModal} testID="closeModalButton">
              <Text style={styles.closeButtonText}>Fechar</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisibleConsumo}
        onRequestClose={toggleModalConsumo}
        testID="modalConsumo"
      >
        <View style={styles.modalContainer}>
          <View style={styles.modalContent}>
            <Text style={styles.modalNome} testID="modalConsumirNome">Consumir {alimento.nome}</Text>
            <View style={styles.pickerContainer} testID="pickerContainer">
              <Picker
                selectedValue={nomeGrupo}
                onValueChange={(itemValue) => {
                  setNomeGrupo(itemValue);
                }}
                testID="grupoPicker"
              >
                <Picker.Item label="Selecione o grupo" value={null} testID="selectGroup" />
                {Object.keys(GruposEnum).map((grupo) => (
                  <Picker.Item
                    key={grupo}
                    label={GruposEnum[grupo as keyof typeof GruposEnum]}
                    value={GruposEnum[grupo as keyof typeof GruposEnum]}
                    testID={`grupoPickerItem_${grupo}`}
                  />
                ))}
              </Picker>
            </View>
            <TextInput
              style={styles.input}
              placeholder="Porção"
              value={porcaoInput}
              onChangeText={setPorcaoInput}
              keyboardType="numeric"
              testID="porcaoInput"
            />
            <TextInput
              style={styles.input}
              placeholder="Quantidade"
              value={quantidadeInput}
              onChangeText={setQuantidadeInput}
              keyboardType="numeric"
              testID="quantidadeInput"
            />
            <View style={styles.buttonContainer}>
              <TouchableOpacity
                style={styles.confirmButton}
                onPress={handleConfirmarConsumo}
                testID="confirmButton"
              >
                <Text style={styles.closeButtonText}>Confirmar</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.closeButton}
                onPress={toggleModalConsumo}
                testID="closeModalConsumoButton"
              >
                <Text style={styles.closeButtonText}>Cancelar</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
};

export default AlimentoItem;
