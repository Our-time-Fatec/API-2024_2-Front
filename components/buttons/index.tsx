import React, { useEffect, useState } from 'react';
import styles from './styles';
import { View, Text, Image, StyleSheet, TouchableOpacity } from 'react-native';
import { useFonts, Poppins_700Bold, Poppins_400Regular} from '@expo-google-fonts/poppins';
import AppLoading from 'expo-app-loading'

interface BotaoAzulProps{
    texto: string;
}

const BotaoAzul: React.FC<BotaoAzulProps> = ({texto}) => {
    const [fontsLoaded] = useFonts({
        Poppins_700Bold,
        Poppins_400Regular
    })
    if(!fontsLoaded){
        <AppLoading />
    }
    return(
        <View>
            <TouchableOpacity style={styles.botaoazul}>
                <Text style={styles.textobotaoazul}>{texto}</Text>
            </TouchableOpacity>
        </View>
    );
};

interface BotaoBrancoProps{
    texto: string;
}

const BotaoBranco: React.FC<BotaoBrancoProps> = ({texto}) => {
    return(
        <View>
            <TouchableOpacity style={styles.botaobranco}>
                <Text style={styles.textobotaobranco}>{texto}</Text>
            </TouchableOpacity>
        </View>
    );
};

export { BotaoAzul, BotaoBranco };