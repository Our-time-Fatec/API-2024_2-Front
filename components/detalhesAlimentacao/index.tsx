import React from 'react';
import { Modal, View, Text, TouchableOpacity } from 'react-native';
import { IDietaSemanal } from '../../interfaces/IGrafico'; // Supondo que o arquivo está no mesmo nível
import { DiasSemana } from '../../enums/diasSemana';
import { styles } from './styles';

interface DiaDetalhesModalProps {
  selectedDay: DiasSemana | null;
  dietaSemanal: IDietaSemanal;
  isVisible: boolean;
  onClose: () => void;
}

const DiaDetalhesModal: React.FC<DiaDetalhesModalProps> = ({ selectedDay, dietaSemanal, isVisible, onClose }) => {
  return (
    <Modal visible={isVisible} transparent={true} animationType="slide">
      <View style={styles.modalContainer}>
        <View style={styles.modalContent}>
          <Text style={styles.modalTitle}>Detalhes do Dia: {selectedDay}</Text>
          <Text style={styles.modalText}>
            Valor Energético: {selectedDay && dietaSemanal[selectedDay]?.total?.valorEnergetico.toFixed(2) || 0} kcal
          </Text>
          <Text style={styles.modalText}>
            Proteínas: {selectedDay && dietaSemanal[selectedDay]?.total?.proteinas.toFixed(2) || 0} g
          </Text>
          <Text style={styles.modalText}>
            Carboidratos: {selectedDay && dietaSemanal[selectedDay]?.total?.carboidratos.toFixed(2) || 0} g
          </Text>
          <Text style={styles.modalText}>
            Fibras: {selectedDay && dietaSemanal[selectedDay]?.total?.fibras.toFixed(2) || 0} g
          </Text>
          <Text style={styles.modalText}>
            Lipídios: {selectedDay && dietaSemanal[selectedDay]?.total?.lipidios.toFixed(2) || 0} g
          </Text>
          <TouchableOpacity style={styles.closeButton} onPress={onClose}>
            <Text style={styles.closeButtonText}>Fechar</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};

export default DiaDetalhesModal;
