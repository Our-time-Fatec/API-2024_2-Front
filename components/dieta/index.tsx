import React, { useState } from "react";
import {
  View,
  Text,
  Modal,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from "react-native";
import { IDietaFixa } from "../../interfaces/IDieta";
import { styles } from "./styles";

interface DietaProps {
  dieta: IDietaFixa;
  isUserDieta: boolean;
  onEdit: (id: string) => void;
  onDelete: (id: string) => void;
}

const DietaItem: React.FC<DietaProps> = ({
  dieta,
  isUserDieta,
  onEdit,
  onDelete,
}) => {
  const [modalVisible, setModalVisible] = useState(false);

  const toggleModal = () => {
    setModalVisible(!modalVisible);
  };

  const formatDate = (date: Date | undefined) => {
    return date
      ? new Date(date).toLocaleDateString("pt-BR")
      : "Data não disponível";
  };

  return (
    <View style={styles.card}>
      <TouchableOpacity onPress={toggleModal}>
        <View style={styles.detailsContainer}>
          <Text style={styles.nome}>Dia: {dieta.diaSemana}</Text>
          <Text style={styles.info}>
            Calorias:{" "}
            {dieta.detalhes?.valorEnergetico
              ? dieta.detalhes.valorEnergetico.toFixed(2)
              : "N/A"}{" "}
            kcal
          </Text>
          <Text style={styles.info}>
            Data de criação:{" "}
            {dieta.criadoEm
              ? formatDate(dieta.criadoEm)
              : "Data não disponível"}
          </Text>
        </View>
      </TouchableOpacity>
      {isUserDieta && (
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.editButton}
            disabled={!dieta._id}
            onPress={() => onEdit(dieta._id ? dieta._id : "")}
          >
            <Text style={styles.closeButtonText}>Editar</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.deleteButton}
            onPress={() => onDelete(dieta._id ? dieta._id : "")}
          >
            <Text style={styles.closeButtonText}>Remover</Text>
          </TouchableOpacity>
        </View>
      )}
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={toggleModal}
      >
        <View style={styles.modalContainer}>
          <View style={styles.modalContent}>
            <ScrollView contentContainerStyle={styles.scrollViewContent}>
              <Text style={styles.modalNome}>Dia: {dieta.diaSemana}</Text>
              <Text style={styles.modalInfo}>
                Data de criação: {formatDate(dieta.criadoEm)}
              </Text>
              <Text style={styles.modalNome}>Detalhes</Text>
              <Text style={styles.modalInfo}>
                Valor energético: {dieta.detalhes?.valorEnergetico.toFixed(2)}{" "}
                Kcal
              </Text>
              <Text style={styles.modalInfo}>
                Proteínas: {dieta.detalhes?.proteinas.toFixed(2)} g
              </Text>
              <Text style={styles.modalInfo}>
                Carboidratos: {dieta.detalhes?.carboidratos.toFixed(2)} g
              </Text>
              <Text style={styles.modalInfo}>
                Fibras: {dieta.detalhes?.fibras.toFixed(2)} g
              </Text>
              <Text style={styles.modalInfo}>
                Lipídios: {dieta.detalhes?.lipidios.toFixed(2)} g
              </Text>
              <Text style={styles.modalNome}>Grupos</Text>

              {dieta.grupos.map((grupo, index) => (
                <View
                  style={styles.minimodalContent}
                  key={grupo._id ? grupo._id : `grupo-${index}`}
                >
                  <Text style={styles.minimodalNome}>{grupo.nome}</Text>
                  {grupo.alimentos.map((alimento, alimentoIndex) => (
                    <Text
                      key={
                        alimento.alimentoId
                          ? alimento.alimentoId
                          : `alimento-${alimentoIndex}`
                      }
                      style={styles.minimodalInfo}
                    >
                      {alimento.quantidade}x - {alimento.nome} -{" "}
                      {alimento.porcao.toString()}g - Total:{" "}
                      {alimento.detalhes?.valorEnergetico.toFixed(2)} Kcal
                    </Text>
                  ))}
                </View>
              ))}
            </ScrollView>
            <TouchableOpacity style={styles.closeButton} onPress={toggleModal}>
              <Text style={styles.closeButtonText}>Fechar</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    </View>
  );
};

export default DietaItem;