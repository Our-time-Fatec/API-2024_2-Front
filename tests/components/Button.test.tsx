import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { BotaoAzul, BotaoBranco } from '../../components/buttons';
import { TouchableOpacity } from 'react-native';

// Mock para a função useFonts
jest.mock('@expo-google-fonts/poppins', () => ({
  useFonts: jest.fn(() => [true]), // Simula que as fontes estão carregadas
}));

// Mock para o SplashScreen
jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

describe("Componentes de botões", () => {
    describe('BotaoAzul', () => {
        it('deve renderizar corretamente', () => {
          const tree = renderer.create(<BotaoAzul texto="Clique aqui" onPress={jest.fn()} />).toJSON();
          expect(tree).toMatchSnapshot(); // Verifica se a renderização corresponde ao snapshot
        });
      
        it('deve chamar a função onPress ao ser pressionado', () => {
          const onPressMock = jest.fn();
          const component = renderer.create(<BotaoAzul texto="Clique aqui" onPress={onPressMock} />);
      
          act(() => {
            // Simula pressionar o botão
            component.root.findByType(TouchableOpacity).props.onPress();
          });
      
          expect(onPressMock).toHaveBeenCalledTimes(1);
        });
      });
      
      describe('BotaoBranco', () => {
        it('deve renderizar corretamente', () => {
          const tree = renderer.create(<BotaoBranco texto="Clique aqui" onPress={jest.fn()} />).toJSON();
          expect(tree).toMatchSnapshot(); // Verifica se a renderização corresponde ao snapshot
        });
      
        it('deve chamar a função onPress ao ser pressionado', () => {
          const onPressMock = jest.fn();
          const component = renderer.create(<BotaoBranco texto="Clique aqui" onPress={onPressMock} />);
      
          act(() => {
            // Simula pressionar o botão
            component.root.findByType(TouchableOpacity).props.onPress();
          });
      
          expect(onPressMock).toHaveBeenCalledTimes(1);
        });
      });      
})