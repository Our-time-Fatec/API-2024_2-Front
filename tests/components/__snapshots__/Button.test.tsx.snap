// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componentes de botões BotaoAzul deve renderizar corretamente 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#407CE2",
        "borderRadius": 40,
        "opacity": 1,
        "paddingHorizontal": 100,
        "paddingVertical": 14,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins_700Bold",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`Componentes de botões BotaoBranco deve renderizar corretamente 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#407CE2",
        "borderRadius": 40,
        "borderWidth": 2,
        "marginTop": 15,
        "opacity": 1,
        "paddingHorizontal": 60,
        "paddingVertical": 14,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#407CE2",
            "fontFamily": "Poppins_700Bold",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;
